examples/v1beta1/early-stopping/median-stop-with-json-format.yaml

```diff
@@ -59,6 +59,9 @@ spec:
       kind: Job
       spec:
         template:
+          metadata:
+            annotations:
+              sidecar.istio.io/inject: "false"
```

Comment:
I agree with @andreyvelich.
It might be better to run the `kubectl patch` command in `run-e2e-experiment.sh`.

Reply:
Actually, as per [conformance test design ](https://docs.google.com/document/d/1TRUKUY1zCCMdgF-nJ7QtzRwifsoQop0V8UnRo-GWlpI/edit?pli=1#heading=h.tubso4j7bqvw), we want to run these tests inside a pod. I don't think we will have access to `kubectl` inside the pod. Correct me if I'm wrong.

We will need to add support for annotations in `run-e2e-experiment.py`

Reply:
I think programmatically patching the resources will not help here as the resources are already created with Istio sidecar.

We will need to update the definition somehow before creating the experiment. This is tricky. If we provided an option to pass annotation, this can create confusion as the passed annotation can be applied in more than one place (experiment metadata, pod metadata, etc).

Reply:
I think, @tenzen-y meant not use `kubectl patch`, but change the run-e2e-experiment script to modify Trial template before creating Experiment. Similar as [here for Trial spec](https://github.com/kubeflow/katib/blob/master/test/e2e/v1beta1/scripts/gh-actions/run-e2e-experiment.py#L259-L261).

Reply:
> I think, @tenzen-y meant not use kubectl patch, but change the run-e2e-experiment script to modify Trial template before creating Experiment. Similar as [here for Trial spec](https://github.com/kubeflow/katib/blob/master/test/e2e/v1beta1/scripts/gh-actions/run-e2e-experiment.py#L259-L261).

Yes, that's right. Sorry for your confusion.

Reply:
@andreyvelich It is bit tricky. Trial Spec can be different. Say distributed job vs non distributed job.

Reply:
@johnugeorge That's right, but since we do it only for Random experiment which uses Kubernetes Job, should we try to `deserialize` TrialSpec to `V1Job` ?

Reply:
I'm ok.

Reply:
@johnugeorge @tenzen-y Other solution might be to add another experiment just for conformance test here: https://github.com/kubeflow/katib/tree/master/test/e2e/v1beta1/testdata with required annotation and other specification.

Reply:
If we add another experiment, I think we should run the experiment on CI every PR similar to other examples.
Since we need to verify whether the experiment can be runnable.

Reply:
@tenzen-y  I see the same exact problem. Conformance tests should be tested regularly. Else, it will diverge from the core examples.  I would suggest to go with current **random** experiment. For annotations, we have to handle 3 cases- non distributed trials, distributed trials and custom trials. We are handling just non distributed trials now. Others can be handled later.